### DISCUSSION TITLE
Draw an hemisphere on Rviz - bumper simulation

### DISCUSSION THREAD

**Discussion by TitouanTyack** [ctu-mrs/mrs_uav_system#102] [Q&A] [answered]
Hello,

I want to visualize a tube which represents the volume which contains the possible position of a drone. The tube is made by one cylinder and two hemispheres. For now, I'm able to visualize it with one cylinder and two spheres. I don't know the easiest way to make hemisphere on Rviz.

![Screenshot from 2021-08-09 15-32-23](https://user-images.githubusercontent.com/85453067/128714924-087244a3-9a7c-4869-8209-e0be4df44c6e.png)

I launched your bumper simulation, and I saw that you managed to visualize the obstacles sectors with none primitive shapes of Rviz (or not entire shapes, it looks like cut cylinders).
So, do you have any idea or information on how can I do a marker of an hemisphere on Rviz, the hemisphere will move and his radius could change.
If you need more precise explanation about my question you can ask. 

Thank you,
Titouan

**Comment by TitouanTyack** [comment, marked as answer]
Hello,

Finaly I succeeded by using the Rviz mesh resource function for a marker with an .stl file of an hemisphere.
![129002442-c341fb7f-8f1d-4dad-9209-841645c4efc0](https://user-images.githubusercontent.com/85453067/129018230-9d21406d-9848-4f2d-9cca-7d4357da9a47.png)

Here you can see the static hemisphere, I am able to change the scale, position and the orientation of the hemisphere in real time (usual stuff of a marker) so it is perfect for me.

Best,
Titouan

**Reply by matemat13** [reply]
Hello! Sorry for the late answer, I didn't get any notification about your question, sadly. If this is still relevant, you can take a look at our implementation of the custom Rviz marker here: https://github.com/ctu-mrs/mrs_rviz_plugins/tree/master/src/bumper (or, more specifically, here: https://github.com/ctu-mrs/mrs_rviz_plugins/blob/master/src/bumper/visual.cpp#L88]
There are also other custom Rviz plugins in that repo in case you're interested.
